Source/WebCore/rendering/RenderText.cpp

```diff
@@ -544,11 +544,10 @@ static Vector<LayoutRect> characterRects(const InlineIterator::TextBox& run, uns
         return { };
 
     if (auto* svgTextBox = dynamicDowncast<SVGInlineTextBox>(run.legacyInlineBox())) {
-        Vector<LayoutRect> rects;
-        rects.reserveInitialCapacity(clampedEnd - clampedStart);
-        for (auto index = clampedStart; index < clampedEnd; ++index)
-            rects.uncheckedAppend(svgTextBox->localSelectionRect(index, index + 1));
-        return rects;
+        return Vector<LayoutRect>(clampedEnd - clampedStart, [&, clampedStart = clampedStart](size_t i) {
```

Comment:
Why do we need to write `clampedStart = clampedStart`? Doesn’t `&` already capture `clampedStart`?

Reply:
It is because `clampedStart` is initialized like so (structured bindings):
```
auto [clampedStart, clampedEnd] = run.selectableRange().clamp(rangeStart, rangeEnd);
```

I was getting a build error until I captured it explicitly by value.

Reply:
To clarify, it is building locally for me with just [&] and that's what I originally had. However, quite a few of the bots didn't like it (both iOS and macOS):
```
/Volumes/Data/worker/macOS-AppleSilicon-Ventura-Debug-Build-EWS/build/Source/WebCore/rendering/RenderText.cpp:548:28: error: reference to local binding 'clampedStart' declared in enclosing function 'WebCore::characterRects'
```

Reply:
Then I think the correct way to say this is something like "`clampedStart = clampedStart` is a workaround for what seems to be a compiler bug in older versions of `clang`".

Reply:
I did some research and found https://en.cppreference.com/w/cpp/language/structured_binding:
```
Structured bindings cannot be captured by lambda expressions (until C++20)
```

So it seems this is only supported in C++20 and it seems some of our toolchains don't support this particular C++20 feature yet.